PathPlanning/BidirectionalAStar/bidirectional_a_star.py

```diff
@@ -0,0 +1,335 @@
+
+            c_id_B = min(
+                open_set_B,
+                key=lambda o: open_set_B[o].cost + self.calc_heuristic(current_A,
+                                                                     open_set_B[
+                                                                         o]))
+
+            current_B = open_set_B[c_id_B]
+
+            # show graph
+            if show_animation:  # pragma: no cover
+                plt.plot(self.calc_grid_position(current_A.x, self.minx),
+                         self.calc_grid_position(current_A.y, self.miny), "xc")
+                plt.plot(self.calc_grid_position(current_B.x, self.minx),
+                         self.calc_grid_position(current_B.y, self.miny), "xc")
+                # for stopping simulation with the esc key.
+                plt.gcf().canvas.mpl_connect('key_release_event',
+                                             lambda event: [exit(
+                                                 0) if event.key == 'escape' else None])
+                if len(closed_set_A.keys()) % 10 == 0:
+                    plt.pause(0.001)
+
+            if current_A.x == current_B.x and current_A.y == current_B.y:
+                print("Found goal")
+                meetpointA = current_A
+                meetpointB = current_B
+                break
+
+            # Remove the item from the open set
+            del open_set_A[c_id_A]
+            del open_set_B[c_id_B]
+
+            # Add it to the closed set
+            closed_set_A[c_id_A] = current_A
+            closed_set_B[c_id_B] = current_B
+
+            # expand_grid search grid based on motion model
+            for i, _ in enumerate(self.motion):
+                continue_A = False
+                continue_B = False
+
+                child_node_A = self.Node(current_A.x + self.motion[i][0],
+                                 current_A.y + self.motion[i][1],
+                                 current_A.cost + self.motion[i][2], c_id_A)
+
+                child_node_B = self.Node(current_B.x + self.motion[i][0],
+                                 current_B.y + self.motion[i][1],
+                                 current_B.cost + self.motion[i][2], c_id_B)
+
+                n_id_A = self.calc_grid_index(child_node_A)
+                n_id_B = self.calc_grid_index(child_node_B)
+
+                # If the node is not safe, do nothing
+                if not self.verify_node(child_node_A):
+                    continue_A = True
+                
+                if not self.verify_node(child_node_B):
+                    continue_B = True
+
+                if n_id_A in closed_set_A:
+                    continue_A = True
+                
+                if n_id_B in closed_set_B:
+                    continue_B = True
+
+                if not(continue_A):
+                    if n_id_A not in open_set_A:
+                        open_set_A[n_id_A] = child_node_A  # discovered a new node
```

Comment:
pep8 too long line.

Reply:
Oops forgot to parse with autopep8, done now.